lib/src/native/realm_core.dart

```diff
@@ -225,7 +227,10 @@ class _RealmCore {
       } else if (config is DisconnectedSyncConfiguration) {
         _realmLib.realm_config_set_force_sync_history(configPtr, true);
       }
-
+      if (config.encryptionKey != null) {
+        assert(config is! InMemoryConfiguration, "Encryption keys are not allowed for InMemoryConfiguration");
```

Comment:
This check is debug only. It will not happen in release builds. Is this the intention?

Reply:
Yes, it is by purpose. Actually, there is no way the user to do that, because there is no `encryptionKey ` parameter in `Configuration.inMemory` method.

Reply:
I suggest we remove this then. Core will also ignore this if its an inmemory one

Reply:
I think asserting is valuable here as it clarifies the behavior. The type system should prevent setting this value, but someone 2 years from now may decide to refactor the configurations and accidentally leak the encryption key into the in memory config and hit that assert.

Reply:
how can you leak a non existing property. Also if someone does that it will still not work in Core. I think we don't need to litter the code with such extra sanity checks which can never happen cause of the API shape.  
Having asserts like that makes sense only if there is a possibility in the API to pass an invalid argument into a function.

Reply:
Yes that is my point. If we can not set through the API, then the `assert` is meaningless. unlike a sanity check for a value which might be returned by a function. And even if we set it in the future unintentionally by a refactoring we what we are doing is duplicating code that is already in Core.

Reply:
I will remove the assert.